core/shared/src/main/scala/scalaz/zio/ZIO.scala

```diff
@@ -652,8 +653,8 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
   /**
    * Keeps some of the errors, and terminates the fiber with the rest.
    */
-  final def refineOrDie[E1](pf: PartialFunction[E, E1])(implicit ev: E <:< Throwable): ZIO[R, E1, A] =
-    refineOrDieWith(pf)(ev)
+  final def refineOrDie[E1: ClassTag](implicit ev: E <:< Throwable): ZIO[R, E1, A] =
```

Comment:
Can we introduce a new method here rather than overwriting the old one? The other one is really convenient when wanting to remap errors and pull out a few different types of errors.

I'd suggest:

```scala
final def refineToOrDie[E1: ClassTag](implicit ev: E <:< Throwable): ZIO[R, E1, A]
```

Reply:
Sure thing! Done 👍